lib/Parse/ParseExpr.cpp

```diff
@@ -2908,6 +2921,37 @@ ExprResult Parser::ParseBoundsExpression() {
   return Result;
 }
 
+ExprResult Parser::ParseRelativeBoundsExpression(Expr *Expr) {
+  ExprResult Result;
+  tok::TokenKind TokKind = Tok.getKind();
+  if (TokKind != tok::identifier) {
```

Comment:
I think you should delete this if-statement.    It would be incorrect say that we are expecting a bounds expression here.  We are expecting a relative bounds expression, which is covered by the next test.